pkg/updater/telemetry.go

```diff
@@ -0,0 +1,120 @@
+package updater
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/gorilla/mux"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/config/utils"
+	"github.com/DataDog/datadog-agent/pkg/internaltelemetry"
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	traceconfig "github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+)
+
+const (
+	telemetryEndpoint = "/v0.4/traces"
+)
+
+type telemetry struct {
+	client internaltelemetry.Client
+
+	site       string
+	socketPath string
+	listener   net.Listener
+	server     *http.Server
+}
+
+func newTelemetry(config config.Reader, socketDirectory string) (*telemetry, error) {
+	socketPath := filepath.Join(socketDirectory, "telemetry.sock")
```

Comment:
a little comment explaining the internal socket hack to get go tracer traces would be nice for future readers